src/Language/Haskell/Liquid/Transforms/Rewrite.hs

```diff
@@ -149,6 +150,10 @@ rewriteWith tx           = go
 
 -}
 
+safeSimplifyPatTuple :: RewriteRule
```

Comment:
I think this is a lot clearer, if more verbose:

```haskell

safeSimplifyPatTuple :: RewriteRule
safeSimplifyPatTuple e
  | Just e' <- simplifyPatTuple e
  , CoreUtils.exprType e' == CoreUtils.exprType e
  = Just e'
  | otherwise
  = Nothing
```